src/checkboxes.ts

```diff
@@ -0,0 +1,203 @@
+'use strict';
+
+import { window, workspace, TextLine, Range, TextEditor, TextEditorEdit } from 'vscode';
+
+// Checkbox is represented by exactly one symbol between square brackets.  Symbol indicates status: '-' undetermined, 'x' or 'X' checked, ' ' unchecked.
+const checkboxRegex = /\[([-xX ])\]/;
+// Summary is a cookie indicating the number of ticked checkboxes in the child list relative to the total number of checkboxes in the list. 
+const summaryRegex = /\[(\d*\/\d*)\]/;
+// Percentage is a cookie indicating the percentage of ticked checkboxes in the child list relative to the total number of checkboxes in the list. 
+const percentRegex = /\[(\d*)%\]/;
+const indentRegex = /^(\s*)\S/;
+let orgTabSize: number = 4;
```

Comment:
Is there a reason to store `orgTabSize` globally. It can be changed on fly. And potentially break traversing via trees. If I understand your logic correctly (sorry if I'm wrong, just made fast review yet and not JS developer).